### PR TITLE
discovery: Move static ignore list to system-probe

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Move the static list of ignored process names to system-probe, so that system-probe can avoid doing unnecessary work.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-1214

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/module/impl_linux.go [hunk 1]
```diff
@@ -353,13 +353,35 @@ func customNewProcess(pid int32) (*process.Process, error) {
...
 	}
 
+	comm, err := proc.Name()
+	if err != nil {
+		return nil
+	}
+
+	if _, found := ignoreComms[comm]; found {
+		return nil
+	}
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/module/impl_linux.go:383]: The current code will work as all entries in `ignoreComms` are less than  or equal 16 characters
I do suspect we might add another entry to `ignoreComms` in the future, and we will not be aware of the assumption of 16 characters
I do suggest adding a comment & a UT to verify the entries length does not exceed 16 characters

**Review by guyarb — COMMENTED** [review]
Did you verify the output of the check (and not the http endpoint of system-probe) does not report the ignored services?

File: pkg/collector/corechecks/servicediscovery/module/impl_linux.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/collector/corechecks/servicediscovery/module/impl_linux.go:383]: Done.

**Comment by vitkyrka** [comment]
> Did you verify the output of the check (and not the http endpoint of system-probe) does not report the ignored services?

Yes.

**Review by guyarb — APPROVED** [review]

**Pull request merged**